test/fixtures/test-runner/output/name_pattern_with_only.js

```diff
@@ -5,7 +5,7 @@ const { test } = require('node:test');
 
 test('enabled and only', { only: true }, common.mustCall(async (t) => {
   await t.test('enabled', common.mustCall());
-  await t.test('disabled', common.mustNotCall());
+  await t.test('disabled but parent not', common.mustCall());
```

Comment:
Why is this running? Is it because the parent is enabled? If so, I don't think that's what https://github.com/nodejs/node/issues/46728 is about.

Reply:
see https://github.com/nodejs/node/pull/48382#issuecomment-1583918047

Reply:
FWIW, this change corresponds to my expectations.
A more realistic use case:

```js
describe('add', () => {
  it('should return 2', () => assert.strictEqual(add(1, 1), 2));
  it('should return 10', () => assert.strictEqual(add(8, 2), 10));
  it('should throw', () => assert.throws(add(), TypeError));
});

describe('subtract', () => {
  // ...
});
```

`node --test-name-pattern=add` should run all children of `describe('add')`

Reply:
I'm fine with this PR now that it's clear this is not fixing #46728.

Reply:
Can you please approve? :)